### PR TITLE
Update the CodeQL data flow models

### PR DESCRIPTION
**Pull request by knutwannheden**
Updates the `model.csv` and `sinks.csv` files using the most current CodeQL release 2.12.2.

The queries had to be slightly changed, but the output is basically the same (except for the `generated` column which has been renamed to `provenance` with the values `manual` and `generated`). Also, `codeql query run` outputs BQRS files which can then be transformed to CSV using `codeql bqrs decode`. The resulting CSV files look slightly different (with `,` as separator rather than `;` and strings being quoted).

Please note that the new models contain more data and are presumably more accurate.

### CODE REVIEW

File: rewrite-java/src/main/java/org/openrewrite/java/dataflow/internal/csv/CsvLoader.java [hunk 1]
```diff
@@ -51,4 +51,32 @@ private static <R extends Mergeable<R>, E> R loadCvs(InputStream input, URI sour
...
+            while (i < line.length() && line.charAt(i) != stopAt) {
+                i++;
+            }
+            result.add(line.substring(valueStart, i));
+            if (stopAt == '"') {
+                i++;
+            }
+        }
+        return result.toArray(new String[0]);
+    }
```
Comment (JLLeitschuh) [inline rewrite-java/src/main/java/org/openrewrite/java/dataflow/internal/csv/CsvLoader.java:80]: Ah, okay, interesting. This works I suppose.

File: rewrite-java/src/main/java/org/openrewrite/java/dataflow/internal/csv/CsvLoader.java [hunk 2]
```diff
@@ -51,4 +51,32 @@ private static <R extends Mergeable<R>, E> R loadCvs(InputStream input, URI sour
         }
     }
 
+    private static String[] parseLine(String line) {
+        if (line.isEmpty()) {
+            return new String[0];
+        }
+
+        ArrayList<String> result = new ArrayList<>();
```
Comment (JLLeitschuh) [inline rewrite-java/src/main/java/org/openrewrite/java/dataflow/internal/csv/CsvLoader.java:59]: You could theoretically pre-size this as a max size of 10

Comment (knutwannheden) [reply rewrite-java/src/main/java/org/openrewrite/java/dataflow/internal/csv/CsvLoader.java:59]: Which happens to be the default capacity of an `ArrayList` 😀. If we wanted to optimize this, there would be other means too, such as always reusing the same array (of the correct length). But it's a meager 10K rows, so I didn't bother.

File: rewrite-java/src/main/java/org/openrewrite/java/dataflow/internal/csv/CsvLoader.java (hunk 1 above)
Comment (knutwannheden) [reply rewrite-java/src/main/java/org/openrewrite/java/dataflow/internal/csv/CsvLoader.java:80]: The simple `String#split()` doesn't work anymore due to the quoted values and I didn't want to bring in a new Jackson dependency just for this. But like you noticed it isn't pretty and by no means optimized. Feel free to improve :-)

File: rewrite-java/src/main/java/org/openrewrite/java/dataflow/internal/csv/CsvLoader.java (hunk 2 above)
Comment (JLLeitschuh) [reply rewrite-java/src/main/java/org/openrewrite/java/dataflow/internal/csv/CsvLoader.java:59]: > Which happens to be the default capacity of an ArrayList.

Oh, wow, didn't know this. Thanks! 😆

**Pull request merged**